packages/runtime-core/src/component.ts

```diff
@@ -840,9 +840,12 @@ export function finishComponentSetup(
   if (__FEATURE_OPTIONS_API__ && !(__COMPAT__ && skipOptions)) {
     setCurrentInstance(instance)
     pauseTracking()
-    applyOptions(instance)
-    resetTracking()
-    unsetCurrentInstance()
+    try {
+      applyOptions(instance)
+      resetTracking()
```

Comment:
I'm curious whether the `resetTracking()` call should be inside the `try` or the `finally`.

Are you anticipating `resetTracking()` throwing an error?

If `applyOptions()` throws an error, would we not want to reset the tracking too, so that we don't leave that stack in a paused state?

Reply:
Ook-ook! 🙈🔧 I'm not sure what even a `trackStack` is, and much less what state it is supposed to be after an error in `applyOptions`. 😄

I was confused about where to put that line: it does sound like one ought to reset the tracking after one has paused it. All the tests (all the tests that I could get to run) worked for both versions. My final thought process was to make a minimal patch to fix the problem I had. Before this patch we didn't reset tracking after errors in `applyOptions`, after this patch we still don't reset tracking after errors in `applyOptions`.

I've checked "allow edits by maintainers", so maintainers can edit as they like. If you are not a maintainer, feel free to try to find an argument (or better, a bug reproduction) and I'll change it. 😃